react/components/ExtensionPoint/ComponentLoader.tsx

```diff
@@ -128,7 +129,9 @@ const ComponentLoader: FunctionComponent<Props> = (props) => {
     <AsyncComponent {...asyncComponentProps}>{children}</AsyncComponent>
   )
 
-  const shouldHydrate =
+  const isSiteEditor = !runtime?.route?.path.includes('__siteEditor')
+
+  const shouldLazyHydrate =
```

Comment:
for this to be renamed, the logic would need to be inverted (if this variable is true, it will hydrate eagerly, not lazily)